core/io/file_access_compressed.cpp

```diff
@@ -33,6 +33,11 @@
 #include "core/string/print_string.h"
 
 void FileAccessCompressed::configure(const String &p_magic, Compression::Mode p_mode, uint32_t p_block_size) {
+	if (p_block_size == 0) {
+		// Treat a block size of 0 as solid compression (compress entire file as a single block).
+		// Solid compression favors compression ratio at the cost of performance and memory usage.
+		p_block_size = -1;
```

Comment:
Block size is saved in the file and there's `buffer.resize(block_size);` in the read code, so it would resize the buffer to 4GB, I would probably keep it `0` and add special cases in the rest of the file.

Reply:
This sounds like a lot of work for something that may not be used often. Maybe I should just remove the ability to use solid compression and advise users to use a large block size if they want the best possible compression (e.g. `1048576` bytes for 1 MiB blocks). What do you think?

Reply:
@bruvzg Bump :slightly_smiling_face: